src/db-adapters/Mongoose/MongooseAdapter.js

```diff
@@ -614,4 +596,29 @@ export default class MongooseAdapter {
 
     return words.join(" ");
   }
+
+  static getIdQueryType(idOrIds) {
+    const mode = typeof idOrIds === "string" ? "findOne" : "find";
+    let idQuery;
+
+    if (typeof idOrIds === "string") {
+      if (!this.idIsValid(idOrIds)) {
+        throw new APIError(404, undefined, "No matching resource found.", "Invalid ID.");
+      }
+
+      idQuery = {_id: idOrIds};
+    } else if (Array.isArray(idOrIds)) {
+      if (idOrIds.some(this.idIsValid)) {
```

Comment:
Wouldn't changing it to `.every()` mean it would only throw if every ID is invalid?